### PR TITLE
Add _info / _facts module related notes

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Add _info / _facts module related notes

### CODE REVIEW

**Comment by Andersson007** [comment]
ready_for_review

File: collection_checklist.md [hunk 1]
```diff
@@ -15,7 +15,10 @@ Every comment should say whether the reviewer expects it to be addressed, or whe
 - [ ] adheres to [semantic versioning](https://semver.org/)
 - [ ] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
 - [ ] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
-- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
+- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); besides the other requirements:
+  - [ ] modules that only gather information are named `<something>_info` or `<something>_facts`
+  - [ ] modules do not return information using the `state` option values
```
Comment (felixfontein) [inline collection_checklist.md:20]:
```suggestion
  - [ ] modules do not allow to query information using specific the `state` option values, or similar mechanisms (like `state=get` or `state=query`)
```

File: collection_requirements.rst [hunk 2]
```diff
@@ -341,6 +349,15 @@ Also:
 
 See `Migrating content to a different collection <https://docs.ansible.com/ansible/devel/dev_guide/developing_collections.html#migrating-ansible-content-to-a-different-collection>`_ for complete details.
 
+Development conventions
+=======================
+
+Besides all the requirements listed in the `Development conventions <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html>`_, be sure:
+
+* your modules do not return information using the ``state`` option values like ``get``, ``list``, ``query``, or ``info`` -
```
Comment (felixfontein) [inline collection_requirements.rst:357]:
```suggestion
* your modules do not allow to query information using special ``state`` option values like ``get``, ``list``, ``query``, or ``info`` -
```

Comment (Andersson007) [reply collection_requirements.rst:357]: committed, thanks!

**Comment by felixfontein** [comment]
@Andersson007 LGTM. Can you create an issue so we can discuss it at the next meeting?

**Comment by Andersson007** [comment]
@felixfontein done https://github.com/ansible-community/community-topics/issues/14, thanks!

File: collection_checklist.md [hunk 3]
```diff
@@ -15,7 +15,10 @@ Every comment should say whether the reviewer expects it to be addressed, or whe
 - [ ] adheres to [semantic versioning](https://semver.org/)
 - [ ] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
 - [ ] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
-- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
+- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); besides the other requirements:
+  - [ ] modules that only gather information are named `<something>_info` or `<something>_facts`
+  - [ ] modules do not allow to query information using specific the `state` option values, or similar mechanisms (like `state=get` or `state=query`)
```
Comment (aminvakil) [inline collection_checklist.md:20]:
```suggestion
  - [ ] modules do not allow to query information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`)
```

Comment (Andersson007) [reply collection_checklist.md]: @aminvakil committed, thanks!

File: collection_checklist.md [hunk 4]
```diff
@@ -15,7 +15,10 @@ Every comment should say whether the reviewer expects it to be addressed, or whe
 - [ ] adheres to [semantic versioning](https://semver.org/)
 - [ ] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
 - [ ] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
-- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
+- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); besides the other requirements:
```
Comment (samccann) [inline collection_checklist.md:18]:
```suggestion
- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
```

File: collection_checklist.md [hunk 5]
```diff
@@ -15,7 +15,10 @@ Every comment should say whether the reviewer expects it to be addressed, or whe
...
+  - [ ] modules that only gather information are named `<something>_info` or `<something>_facts`
```
Comment (jillr) [inline collection_checklist.md:19]:
```suggestion
  - [ ] modules that only gather information are named `<something>_info`
  - [] modules that only return `ansible_facts` are named `<something>_facts`
```

File: collection_checklist.md [hunk 6]
```diff
@@ -15,7 +15,10 @@ Every comment should say whether the reviewer expects it to be addressed, or whe
...
+  - [ ] modules do not allow to query information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`)
```
Comment (samccann) [inline collection_checklist.md:20]:
```suggestion
  - [ ] modules do not query information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`)
```

Comment (abadger) [inline collection_checklist.md:20]:
```suggestion
  - [ ] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
```

File: collection_requirements.rst [hunk 7]
```diff
@@ -341,6 +349,15 @@ Also:
 
 See `Migrating content to a different collection <https://docs.ansible.com/ansible/devel/dev_guide/developing_collections.html#migrating-ansible-content-to-a-different-collection>`_ for complete details.
 
+Development conventions
+=======================
+
+Besides all the requirements listed in the `Development conventions <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html>`_, be sure:
+
+* your modules do not allow to query information using special ``state`` option values like ``get``, ``list``, ``query``, or ``info`` -
```
Comment (samccann) [inline collection_requirements.rst:357]:
```suggestion
* Your modules do not query information using special ``state`` option values like ``get``, ``list``, ``query``, or ``info`` -
```

File: collection_checklist.md (hunk 6 above)
Comment (felixfontein) [reply collection_checklist.md:20]: Hmm, modules themselves almost always query stuff. This is more about using the module. I think @abadger's suggestion below is more precise.

File: collection_requirements.rst [hunk 8]
```diff
@@ -215,6 +218,11 @@ We should avoid FQCN / repository names:
 
 If your collection is planned to be certified on Automation Hub, please consult with Red Hat Partner Engineering to ensure collection naming compatibility between the community collection on **Galaxy**.
 
+Module naming
+-------------
+
+Modules that only gather information MUST be named ``<something>_info`` or ``<something>_facts``.
```
Comment (jillr) [inline collection_requirements.rst:224]:
```suggestion
Modules that only gather information MUST be named ``<something>_info``. Modules that only return ``ansible_facts`` are named ``<something>_facts`` and do not return non-facts
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
Thanks @Andersson007 and everyone who helped!

**Pull request merged**